src/arbcall.jl

```diff
@@ -107,17 +165,14 @@ function jlcode(af::Arbfunction, jl_fname=jlfname(af))
 
     jl_args = [:($a::$T) for (a, T) in zip(arg_names, jl_types)]
 
-    res = first(arg_names)
-
     return :(
-        function $jl_fname($(jl_args...); $(kwargs...))
-            ccall(Arblib.@libarb($(arbfname(af))),
-            $returnT,
-            $(Expr(:tuple, c_types...)),
-            $(Symbol.(name.(args))...))
-            return $res
+        function $jl_fname($(jl_args...); $(kwargs...))::$returnT
+        ccall(Arblib.@libarb($(arbfname(af))),
+              $returnT,
+              $(Expr(:tuple, c_types...)),
+              $(Symbol.(name.(args))...))
         end
-        )
+    )
```

Comment:
It would be great if someone can figure out how to use `jlsignature(af)` here. Just
```
return :(
    function $(jlsignature(af))
    ccall(Arblib.@libarb($(arbfname(af))),
          $returnT,
          $(Expr(:tuple, c_types...)),
          $(Symbol.(name.(args))...))
    end
)
```
did not work.

Reply:
to make it work you may need to modify AST directly;
also note that in `jlsignature` (as defined) `x::Arb` means something different (type assertion) than in `function is_zero(x::Arb) end`.

```julia
julia> f = Arbfunction("int arb_is_zero(const arb_t x)");
julia> dump(Arblib.jlsignature(f))
Expr
  head: Symbol ::
  args: Array{Any}((2,))
    1: Expr
      head: Symbol call
      args: Array{Any}((3,))
        1: Symbol is_zero
        2: Expr
          head: Symbol parameters
          args: Array{Any}((0,))
        3: Expr
          head: Symbol ::
          args: Array{Any}((2,))
            1: Symbol x
            2: Arb <: Real
              arb::Arblib.arb_struct
              prec::Int64
    2: Int32 <: Signed

julia> ex = :(function is_zero(x::Arb; )::Int32 end); dump(ex)
Expr
  head: Symbol function
  args: Array{Any}((2,))
    1: Expr
      head: Symbol ::
      args: Array{Any}((2,))
        1: Expr
          head: Symbol call
          args: Array{Any}((3,))
            1: Symbol is_zero
            2: Expr
              head: Symbol parameters
              args: Array{Any}((0,))
            3: Expr
              head: Symbol ::
              args: Array{Any}((2,))
                1: Symbol x
                2: Symbol Arb
        2: Symbol Int32
    2: Expr
      head: Symbol block
      args: Array{Any}((2,))
        1: LineNumberNode
          line: Int64 2
          file: Symbol REPL[7]

```

Reply:
Yes, I had a lot of issues with this trying to write the tests. But okay, I'll take another look and see if I can modify the AST directly.

Reply:
I got it to work, it was not to complicated in the end. The new version does not produce identical AST to the old one, but the result should be the same. I also added some very simple tests to make sure you at least can call the generated methods.